DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/Delegates/AddDelegateToGroupControllerTests.cs

```diff
@@ -0,0 +1,250 @@
+        private CentreCustomPromptHelper centreCustomPromptHelper;
+        private IClockService clockService;
+        private IGroupsDataService groupsDataService;
+        private IGroupsService groupsService;
+        private HttpRequest httpRequest = null!;
+        private HttpResponse httpResponse = null!;
+        private IJobGroupsDataService jobGroupsDataService;
+        private IUserDataService userDataService;
+
+        [SetUp]
+        public void Setup()
+        {
+            centreCustomPromptHelper = A.Fake<CentreCustomPromptHelper>();
+            groupsDataService = A.Fake<IGroupsDataService>();
+            groupsService = A.Fake<IGroupsService>();
+            clockService = A.Fake<IClockService>();
+            jobGroupsDataService = A.Fake<IJobGroupsDataService>();
+            userDataService = A.Fake<IUserDataService>();
+
+            httpRequest = A.Fake<HttpRequest>();
+            httpResponse = A.Fake<HttpResponse>();
+            const string cookieName = "AddDelegateToGroupFilter";
+            const string cookieValue = "ActiveStatus|Active|false";
+
+            A.CallTo(() => jobGroupsDataService.GetJobGroupsAlphabetical()).Returns(
+                JobGroupsTestHelper.GetDefaultJobGroupsAlphabetical()
+            );
+
+            A.CallTo(() => userDataService.GetDelegatesNotRegisteredForGroupByGroupId(A<int>._, A<int>._))
+                .Returns(delegateUserCards);
+
+            A.CallTo(() => userDataService.GetGroupNameById(A<int>._))
+                .Returns("Group name");
+
+            addDelegateToGroupsController = new AddDelegateToGroupController(
+                    jobGroupsDataService,
+                    userDataService,
+                    centreCustomPromptHelper,
+                    groupsService,
+                    groupsDataService,
+                    clockService
+                )
+                .WithMockHttpContext(httpRequest, cookieName, cookieValue, httpResponse)
+                .WithMockUser(true)
+                .WithMockServices()
+                .WithMockTempData();
+        }
+
+        [Test]
+        public void Index_with_no_query_parameters_uses_cookie_value_for_filterBy()
+        {
+            // When
+            var result = addDelegateToGroupsController.Index();
+
+            // Then
+            result.As<ViewResult>().Model.As<AddDelegateToGroupViewModel>().FilterBy.Should()
+                .Be("ActiveStatus|Active|false");
+        }
+
+        [Test]
+        public void Index_with_query_parameters_uses_query_parameter_value_for_filterBy()
+        {
+            // Given
+            const string filterBy = "PasswordStatus|IsPasswordSet|true";
+            A.CallTo(() => httpRequest.Query.ContainsKey("filterBy")).Returns(true);
+
+            // When
+            var result = addDelegateToGroupsController.Index(filterBy: filterBy);
+
+            // Then
+            result.As<ViewResult>().Model.As<AllDelegatesViewModel>().FilterBy.Should()

```

Comment:
Is that the correct view model?

Reply:
Sorry, that's good spot. I have changed it. Thanks